From 64e6a30314469b637169c24cb2aef9c37b461162 Mon Sep 17 00:00:00 2001
From: changeforan <7800290+changeforan@users.noreply.github.com>
Date: Sat, 22 Aug 2026 19:20:39 +0800
Subject: [PATCH 3/5] winevulkan: Enable VK_KHR_video_decode_h265.

The extension adds no new commands, only structures and enumeration
values, so no additional thunking is required.

The StdVideoH265SubLayerHrdParameters structures referenced by
StdVideoH265HrdParameters are layout compatible between 32-bit and
64-bit, and the generator emits no separate 32-bit variant for them.
---
 dlls/winevulkan/make_vulkan     |   1 -
 dlls/winevulkan/vulkan_thunks.c | 564 ++++++++++++++++++++++++++++++++
 include/wine/vulkan.h           |  62 ++++
 3 files changed, 626 insertions(+), 1 deletion(-)

diff --git a/dlls/winevulkan/make_vulkan b/dlls/winevulkan/make_vulkan
index 792f1d95..e4e73d45 100755
--- a/dlls/winevulkan/make_vulkan
+++ b/dlls/winevulkan/make_vulkan
@@ -101,7 +101,6 @@ UNSUPPORTED_EXTENSIONS = [
     "VK_KHR_maintenance7", # Causes infinity recursion in struct convert code
     "VK_KHR_shared_presentable_image", # Needs WSI work.
     "VK_KHR_video_encode_h265", # StdVideoH265HrdParameters cannot be handled
-    "VK_KHR_video_decode_h265", # by struct conversions.
     "VK_KHR_video_maintenance2", # Also affected by StdVideoH265HrdParameters.
     "VK_NV_external_memory_rdma", # Needs shared resources work.
     "VK_NV_external_compute_queue", # Has a new dispatchable handle
diff --git a/dlls/winevulkan/vulkan_thunks.c b/dlls/winevulkan/vulkan_thunks.c
index dc22cf5e..b361707e 100644
--- a/dlls/winevulkan/vulkan_thunks.c
+++ b/dlls/winevulkan/vulkan_thunks.c
@@ -27,6 +27,25 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(vulkan);
 
+typedef struct StdVideoH265HrdParameters32
+{
+    StdVideoH265HrdFlags flags;
+    uint8_t tick_divisor_minus2;
+    uint8_t du_cpb_removal_delay_increment_length_minus1;
+    uint8_t dpb_output_delay_du_length_minus1;
+    uint8_t bit_rate_scale;
+    uint8_t cpb_size_scale;
+    uint8_t cpb_size_du_scale;
+    uint8_t initial_cpb_removal_delay_length_minus1;
+    uint8_t au_cpb_removal_delay_length_minus1;
+    uint8_t dpb_output_delay_length_minus1;
+    uint8_t cpb_cnt_minus1[STD_VIDEO_H265_SUBLAYERS_LIST_SIZE];
+    uint16_t elemental_duration_in_tc_minus1[STD_VIDEO_H265_SUBLAYERS_LIST_SIZE];
+    uint16_t reserved[3];
+    PTR32 pSubLayerHrdParametersNal;
+    PTR32 pSubLayerHrdParametersVcl;
+} StdVideoH265HrdParameters32;
+
 typedef struct VkGeometryAABBNV32
 {
     VkStructureType sType;
@@ -81,6 +100,36 @@ typedef struct StdVideoH264SequenceParameterSetVui32
     PTR32 pHrdParameters;
 } StdVideoH264SequenceParameterSetVui32;
 
+typedef struct StdVideoH265SequenceParameterSetVui32
+{
+    StdVideoH265SpsVuiFlags flags;
+    StdVideoH265AspectRatioIdc aspect_ratio_idc;
+    uint16_t sar_width;
+    uint16_t sar_height;
+    uint8_t video_format;
+    uint8_t colour_primaries;
+    uint8_t transfer_characteristics;
+    uint8_t matrix_coeffs;
+    uint8_t chroma_sample_loc_type_top_field;
+    uint8_t chroma_sample_loc_type_bottom_field;
+    uint8_t reserved1;
+    uint8_t reserved2;
+    uint16_t def_disp_win_left_offset;
+    uint16_t def_disp_win_right_offset;
+    uint16_t def_disp_win_top_offset;
+    uint16_t def_disp_win_bottom_offset;
+    uint32_t vui_num_units_in_tick;
+    uint32_t vui_time_scale;
+    uint32_t vui_num_ticks_poc_diff_one_minus1;
+    uint16_t min_spatial_segmentation_idc;
+    uint16_t reserved3;
+    uint8_t max_bytes_per_pic_denom;
+    uint8_t max_bits_per_min_cu_denom;
+    uint8_t log2_max_mv_length_horizontal;
+    uint8_t log2_max_mv_length_vertical;
+    PTR32 pHrdParameters;
+} StdVideoH265SequenceParameterSetVui32;
+
 typedef struct VkAccelerationStructureGeometryAabbsDataKHR32
 {
     VkStructureType sType;
@@ -361,6 +410,105 @@ typedef struct StdVideoH264SequenceParameterSet32
     PTR32 pSequenceParameterSetVui;
 } StdVideoH264SequenceParameterSet32;
 
+typedef struct StdVideoH265PictureParameterSet32
+{
+    StdVideoH265PpsFlags flags;
+    uint8_t pps_pic_parameter_set_id;
+    uint8_t pps_seq_parameter_set_id;
+    uint8_t sps_video_parameter_set_id;
+    uint8_t num_extra_slice_header_bits;
+    uint8_t num_ref_idx_l0_default_active_minus1;
+    uint8_t num_ref_idx_l1_default_active_minus1;
+    int8_t init_qp_minus26;
+    uint8_t diff_cu_qp_delta_depth;
+    int8_t pps_cb_qp_offset;
+    int8_t pps_cr_qp_offset;
+    int8_t pps_beta_offset_div2;
+    int8_t pps_tc_offset_div2;
+    uint8_t log2_parallel_merge_level_minus2;
+    uint8_t log2_max_transform_skip_block_size_minus2;
+    uint8_t diff_cu_chroma_qp_offset_depth;
+    uint8_t chroma_qp_offset_list_len_minus1;
+    int8_t cb_qp_offset_list[STD_VIDEO_H265_CHROMA_QP_OFFSET_LIST_SIZE];
+    int8_t cr_qp_offset_list[STD_VIDEO_H265_CHROMA_QP_OFFSET_LIST_SIZE];
+    uint8_t log2_sao_offset_scale_luma;
+    uint8_t log2_sao_offset_scale_chroma;
+    int8_t pps_act_y_qp_offset_plus5;
+    int8_t pps_act_cb_qp_offset_plus5;
+    int8_t pps_act_cr_qp_offset_plus3;
+    uint8_t pps_num_palette_predictor_initializers;
+    uint8_t luma_bit_depth_entry_minus8;
+    uint8_t chroma_bit_depth_entry_minus8;
+    uint8_t num_tile_columns_minus1;
+    uint8_t num_tile_rows_minus1;
+    uint8_t reserved1;
+    uint8_t reserved2;
+    uint16_t column_width_minus1[STD_VIDEO_H265_CHROMA_QP_OFFSET_TILE_COLS_LIST_SIZE];
+    uint16_t row_height_minus1[STD_VIDEO_H265_CHROMA_QP_OFFSET_TILE_ROWS_LIST_SIZE];
+    uint32_t reserved3;
+    PTR32 pScalingLists;
+    PTR32 pPredictorPaletteEntries;
+} StdVideoH265PictureParameterSet32;
+
+typedef struct StdVideoH265SequenceParameterSet32
+{
+    StdVideoH265SpsFlags flags;
+    StdVideoH265ChromaFormatIdc chroma_format_idc;
+    uint32_t pic_width_in_luma_samples;
+    uint32_t pic_height_in_luma_samples;
+    uint8_t sps_video_parameter_set_id;
+    uint8_t sps_max_sub_layers_minus1;
+    uint8_t sps_seq_parameter_set_id;
+    uint8_t bit_depth_luma_minus8;
+    uint8_t bit_depth_chroma_minus8;
+    uint8_t log2_max_pic_order_cnt_lsb_minus4;
+    uint8_t log2_min_luma_coding_block_size_minus3;
+    uint8_t log2_diff_max_min_luma_coding_block_size;
+    uint8_t log2_min_luma_transform_block_size_minus2;
+    uint8_t log2_diff_max_min_luma_transform_block_size;
+    uint8_t max_transform_hierarchy_depth_inter;
+    uint8_t max_transform_hierarchy_depth_intra;
+    uint8_t num_short_term_ref_pic_sets;
+    uint8_t num_long_term_ref_pics_sps;
+    uint8_t pcm_sample_bit_depth_luma_minus1;
+    uint8_t pcm_sample_bit_depth_chroma_minus1;
+    uint8_t log2_min_pcm_luma_coding_block_size_minus3;
+    uint8_t log2_diff_max_min_pcm_luma_coding_block_size;
+    uint8_t reserved1;
+    uint8_t reserved2;
+    uint8_t palette_max_size;
+    uint8_t delta_palette_max_predictor_size;
+    uint8_t motion_vector_resolution_control_idc;
+    uint8_t sps_num_palette_predictor_initializers_minus1;
+    uint32_t conf_win_left_offset;
+    uint32_t conf_win_right_offset;
+    uint32_t conf_win_top_offset;
+    uint32_t conf_win_bottom_offset;
+    PTR32 pProfileTierLevel;
+    PTR32 pDecPicBufMgr;
+    PTR32 pScalingLists;
+    PTR32 pShortTermRefPicSet;
+    PTR32 pLongTermRefPicsSps;
+    PTR32 pSequenceParameterSetVui;
+    PTR32 pPredictorPaletteEntries;
+} StdVideoH265SequenceParameterSet32;
+
+typedef struct StdVideoH265VideoParameterSet32
+{
+    StdVideoH265VpsFlags flags;
+    uint8_t vps_video_parameter_set_id;
+    uint8_t vps_max_sub_layers_minus1;
+    uint8_t reserved1;
+    uint8_t reserved2;
+    uint32_t vps_num_units_in_tick;
+    uint32_t vps_time_scale;
+    uint32_t vps_num_ticks_poc_diff_one_minus1;
+    uint32_t reserved3;
+    PTR32 pDecPicBufMgr;
+    PTR32 pHrdParameters;
+    PTR32 pProfileTierLevel;
+} StdVideoH265VideoParameterSet32;
+
 typedef union VkAccelerationStructureGeometryDataKHR32
 {
     VkAccelerationStructureGeometryTrianglesDataKHR32 DECLSPEC_ALIGN(8) triangles;
@@ -1568,6 +1716,18 @@ typedef struct VkVideoDecodeH264SessionParametersAddInfoKHR32
     PTR32 pStdPPSs;
 } VkVideoDecodeH264SessionParametersAddInfoKHR32;
 
+typedef struct VkVideoDecodeH265SessionParametersAddInfoKHR32
+{
+    VkStructureType sType;
+    PTR32 pNext;
+    uint32_t stdVPSCount;
+    PTR32 pStdVPSs;
+    uint32_t stdSPSCount;
+    PTR32 pStdSPSs;
+    uint32_t stdPPSCount;
+    PTR32 pStdPPSs;
+} VkVideoDecodeH265SessionParametersAddInfoKHR32;
+
 typedef struct VkVideoEncodeH264NaluSliceInfoKHR32
 {
     VkStructureType sType;
@@ -9185,6 +9345,46 @@ typedef struct VkVideoDecodeH264SessionParametersCreateInfoKHR32
     PTR32 pParametersAddInfo;
 } VkVideoDecodeH264SessionParametersCreateInfoKHR32;
 
+typedef struct VkVideoDecodeH265CapabilitiesKHR32
+{
+    VkStructureType sType;
+    PTR32 pNext;
+    StdVideoH265LevelIdc maxLevelIdc;
+} VkVideoDecodeH265CapabilitiesKHR32;
+
+typedef struct VkVideoDecodeH265DpbSlotInfoKHR32
+{
+    VkStructureType sType;
+    PTR32 pNext;
+    PTR32 pStdReferenceInfo;
+} VkVideoDecodeH265DpbSlotInfoKHR32;
+
+typedef struct VkVideoDecodeH265PictureInfoKHR32
+{
+    VkStructureType sType;
+    PTR32 pNext;
+    PTR32 pStdPictureInfo;
+    uint32_t sliceSegmentCount;
+    PTR32 pSliceSegmentOffsets;
+} VkVideoDecodeH265PictureInfoKHR32;
+
+typedef struct VkVideoDecodeH265ProfileInfoKHR32
+{
+    VkStructureType sType;
+    PTR32 pNext;
+    StdVideoH265ProfileIdc stdProfileIdc;
+} VkVideoDecodeH265ProfileInfoKHR32;
+
+typedef struct VkVideoDecodeH265SessionParametersCreateInfoKHR32
+{
+    VkStructureType sType;
+    PTR32 pNext;
+    uint32_t maxStdVPSCount;
+    uint32_t maxStdSPSCount;
+    uint32_t maxStdPPSCount;
+    PTR32 pParametersAddInfo;
+} VkVideoDecodeH265SessionParametersCreateInfoKHR32;
+
 typedef struct VkVideoDecodeInfoKHR32
 {
     VkStructureType sType;
@@ -12118,6 +12318,17 @@ static void convert_VkVideoReferenceSlotInfoKHR_win32_to_host(struct conversion_
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_DPB_SLOT_INFO_KHR:
+        {
+            VkVideoDecodeH265DpbSlotInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
+            const VkVideoDecodeH265DpbSlotInfoKHR32 *in_ext = (const VkVideoDecodeH265DpbSlotInfoKHR32 *)in_header;
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_DPB_SLOT_INFO_KHR;
+            out_ext->pNext = NULL;
+            out_ext->pStdReferenceInfo = UlongToPtr(in_ext->pStdReferenceInfo);
+            out_header->pNext = (void *)out_ext;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_ENCODE_AV1_DPB_SLOT_INFO_KHR:
         {
             VkVideoEncodeAV1DpbSlotInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
@@ -13624,6 +13835,19 @@ static void convert_VkVideoDecodeInfoKHR_win32_to_host(struct conversion_context
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PICTURE_INFO_KHR:
+        {
+            VkVideoDecodeH265PictureInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
+            const VkVideoDecodeH265PictureInfoKHR32 *in_ext = (const VkVideoDecodeH265PictureInfoKHR32 *)in_header;
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PICTURE_INFO_KHR;
+            out_ext->pNext = NULL;
+            out_ext->pStdPictureInfo = UlongToPtr(in_ext->pStdPictureInfo);
+            out_ext->sliceSegmentCount = in_ext->sliceSegmentCount;
+            out_ext->pSliceSegmentOffsets = UlongToPtr(in_ext->pSliceSegmentOffsets);
+            out_header->pNext = (void *)out_ext;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_DECODE_VP9_PICTURE_INFO_KHR:
         {
             VkVideoDecodeVP9PictureInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
@@ -15637,6 +15861,17 @@ static void convert_VkVideoProfileInfoKHR_win32_to_host(struct conversion_contex
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PROFILE_INFO_KHR:
+        {
+            VkVideoDecodeH265ProfileInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
+            const VkVideoDecodeH265ProfileInfoKHR32 *in_ext = (const VkVideoDecodeH265ProfileInfoKHR32 *)in_header;
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PROFILE_INFO_KHR;
+            out_ext->pNext = NULL;
+            out_ext->stdProfileIdc = in_ext->stdProfileIdc;
+            out_header->pNext = (void *)out_ext;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_DECODE_USAGE_INFO_KHR:
         {
             VkVideoDecodeUsageInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
@@ -25638,6 +25873,17 @@ static void convert_VkQueryPoolCreateInfo_win32_to_host(struct conversion_contex
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PROFILE_INFO_KHR:
+        {
+            VkVideoDecodeH265ProfileInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
+            const VkVideoDecodeH265ProfileInfoKHR32 *in_ext = (const VkVideoDecodeH265ProfileInfoKHR32 *)in_header;
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PROFILE_INFO_KHR;
+            out_ext->pNext = NULL;
+            out_ext->stdProfileIdc = in_ext->stdProfileIdc;
+            out_header->pNext = (void *)out_ext;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_DECODE_USAGE_INFO_KHR:
         {
             VkVideoDecodeUsageInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
@@ -27614,6 +27860,276 @@ static const VkVideoDecodeH264SessionParametersAddInfoKHR *convert_VkVideoDecode
     return out;
 }
 
+static void convert_StdVideoH265HrdParameters_win32_to_host(const StdVideoH265HrdParameters32 *in, StdVideoH265HrdParameters *out)
+{
+    if (!in) return;
+
+    out->flags = in->flags;
+    out->tick_divisor_minus2 = in->tick_divisor_minus2;
+    out->du_cpb_removal_delay_increment_length_minus1 = in->du_cpb_removal_delay_increment_length_minus1;
+    out->dpb_output_delay_du_length_minus1 = in->dpb_output_delay_du_length_minus1;
+    out->bit_rate_scale = in->bit_rate_scale;
+    out->cpb_size_scale = in->cpb_size_scale;
+    out->cpb_size_du_scale = in->cpb_size_du_scale;
+    out->initial_cpb_removal_delay_length_minus1 = in->initial_cpb_removal_delay_length_minus1;
+    out->au_cpb_removal_delay_length_minus1 = in->au_cpb_removal_delay_length_minus1;
+    out->dpb_output_delay_length_minus1 = in->dpb_output_delay_length_minus1;
+    memcpy(out->cpb_cnt_minus1, in->cpb_cnt_minus1, STD_VIDEO_H265_SUBLAYERS_LIST_SIZE * sizeof(uint8_t));
+    memcpy(out->elemental_duration_in_tc_minus1, in->elemental_duration_in_tc_minus1, STD_VIDEO_H265_SUBLAYERS_LIST_SIZE * sizeof(uint16_t));
+    memcpy(out->reserved, in->reserved, 3 * sizeof(uint16_t));
+    out->pSubLayerHrdParametersNal = UlongToPtr(in->pSubLayerHrdParametersNal);
+    out->pSubLayerHrdParametersVcl = UlongToPtr(in->pSubLayerHrdParametersVcl);
+}
+
+static const StdVideoH265HrdParameters *convert_StdVideoH265HrdParameters_array_win32_to_host(struct conversion_context *ctx, const StdVideoH265HrdParameters32 *in, uint32_t count)
+{
+    StdVideoH265HrdParameters *out;
+    unsigned int i;
+
+    if (!in || !count) return NULL;
+
+    out = conversion_context_alloc(ctx, count * sizeof(*out));
+    for (i = 0; i < count; i++)
+    {
+        convert_StdVideoH265HrdParameters_win32_to_host(&in[i], &out[i]);
+    }
+
+    return out;
+}
+
+static void convert_StdVideoH265VideoParameterSet_win32_to_host(struct conversion_context *ctx, const StdVideoH265VideoParameterSet32 *in, StdVideoH265VideoParameterSet *out)
+{
+    if (!in) return;
+
+    out->flags = in->flags;
+    out->vps_video_parameter_set_id = in->vps_video_parameter_set_id;
+    out->vps_max_sub_layers_minus1 = in->vps_max_sub_layers_minus1;
+    out->reserved1 = in->reserved1;
+    out->reserved2 = in->reserved2;
+    out->vps_num_units_in_tick = in->vps_num_units_in_tick;
+    out->vps_time_scale = in->vps_time_scale;
+    out->vps_num_ticks_poc_diff_one_minus1 = in->vps_num_ticks_poc_diff_one_minus1;
+    out->reserved3 = in->reserved3;
+    out->pDecPicBufMgr = UlongToPtr(in->pDecPicBufMgr);
+    out->pHrdParameters = convert_StdVideoH265HrdParameters_array_win32_to_host(ctx, (const StdVideoH265HrdParameters32 *)UlongToPtr(in->pHrdParameters), 1);
+    out->pProfileTierLevel = UlongToPtr(in->pProfileTierLevel);
+}
+
+static const StdVideoH265VideoParameterSet *convert_StdVideoH265VideoParameterSet_array_win32_to_host(struct conversion_context *ctx, const StdVideoH265VideoParameterSet32 *in, uint32_t count)
+{
+    StdVideoH265VideoParameterSet *out;
+    unsigned int i;
+
+    if (!in || !count) return NULL;
+
+    out = conversion_context_alloc(ctx, count * sizeof(*out));
+    for (i = 0; i < count; i++)
+    {
+        convert_StdVideoH265VideoParameterSet_win32_to_host(ctx, &in[i], &out[i]);
+    }
+
+    return out;
+}
+
+static void convert_StdVideoH265SequenceParameterSetVui_win32_to_host(struct conversion_context *ctx, const StdVideoH265SequenceParameterSetVui32 *in, StdVideoH265SequenceParameterSetVui *out)
+{
+    if (!in) return;
+
+    out->flags = in->flags;
+    out->aspect_ratio_idc = in->aspect_ratio_idc;
+    out->sar_width = in->sar_width;
+    out->sar_height = in->sar_height;
+    out->video_format = in->video_format;
+    out->colour_primaries = in->colour_primaries;
+    out->transfer_characteristics = in->transfer_characteristics;
+    out->matrix_coeffs = in->matrix_coeffs;
+    out->chroma_sample_loc_type_top_field = in->chroma_sample_loc_type_top_field;
+    out->chroma_sample_loc_type_bottom_field = in->chroma_sample_loc_type_bottom_field;
+    out->reserved1 = in->reserved1;
+    out->reserved2 = in->reserved2;
+    out->def_disp_win_left_offset = in->def_disp_win_left_offset;
+    out->def_disp_win_right_offset = in->def_disp_win_right_offset;
+    out->def_disp_win_top_offset = in->def_disp_win_top_offset;
+    out->def_disp_win_bottom_offset = in->def_disp_win_bottom_offset;
+    out->vui_num_units_in_tick = in->vui_num_units_in_tick;
+    out->vui_time_scale = in->vui_time_scale;
+    out->vui_num_ticks_poc_diff_one_minus1 = in->vui_num_ticks_poc_diff_one_minus1;
+    out->min_spatial_segmentation_idc = in->min_spatial_segmentation_idc;
+    out->reserved3 = in->reserved3;
+    out->max_bytes_per_pic_denom = in->max_bytes_per_pic_denom;
+    out->max_bits_per_min_cu_denom = in->max_bits_per_min_cu_denom;
+    out->log2_max_mv_length_horizontal = in->log2_max_mv_length_horizontal;
+    out->log2_max_mv_length_vertical = in->log2_max_mv_length_vertical;
+    out->pHrdParameters = convert_StdVideoH265HrdParameters_array_win32_to_host(ctx, (const StdVideoH265HrdParameters32 *)UlongToPtr(in->pHrdParameters), 1);
+}
+
+static const StdVideoH265SequenceParameterSetVui *convert_StdVideoH265SequenceParameterSetVui_array_win32_to_host(struct conversion_context *ctx, const StdVideoH265SequenceParameterSetVui32 *in, uint32_t count)
+{
+    StdVideoH265SequenceParameterSetVui *out;
+    unsigned int i;
+
+    if (!in || !count) return NULL;
+
+    out = conversion_context_alloc(ctx, count * sizeof(*out));
+    for (i = 0; i < count; i++)
+    {
+        convert_StdVideoH265SequenceParameterSetVui_win32_to_host(ctx, &in[i], &out[i]);
+    }
+
+    return out;
+}
+
+static void convert_StdVideoH265SequenceParameterSet_win32_to_host(struct conversion_context *ctx, const StdVideoH265SequenceParameterSet32 *in, StdVideoH265SequenceParameterSet *out)
+{
+    if (!in) return;
+
+    out->flags = in->flags;
+    out->chroma_format_idc = in->chroma_format_idc;
+    out->pic_width_in_luma_samples = in->pic_width_in_luma_samples;
+    out->pic_height_in_luma_samples = in->pic_height_in_luma_samples;
+    out->sps_video_parameter_set_id = in->sps_video_parameter_set_id;
+    out->sps_max_sub_layers_minus1 = in->sps_max_sub_layers_minus1;
+    out->sps_seq_parameter_set_id = in->sps_seq_parameter_set_id;
+    out->bit_depth_luma_minus8 = in->bit_depth_luma_minus8;
+    out->bit_depth_chroma_minus8 = in->bit_depth_chroma_minus8;
+    out->log2_max_pic_order_cnt_lsb_minus4 = in->log2_max_pic_order_cnt_lsb_minus4;
+    out->log2_min_luma_coding_block_size_minus3 = in->log2_min_luma_coding_block_size_minus3;
+    out->log2_diff_max_min_luma_coding_block_size = in->log2_diff_max_min_luma_coding_block_size;
+    out->log2_min_luma_transform_block_size_minus2 = in->log2_min_luma_transform_block_size_minus2;
+    out->log2_diff_max_min_luma_transform_block_size = in->log2_diff_max_min_luma_transform_block_size;
+    out->max_transform_hierarchy_depth_inter = in->max_transform_hierarchy_depth_inter;
+    out->max_transform_hierarchy_depth_intra = in->max_transform_hierarchy_depth_intra;
+    out->num_short_term_ref_pic_sets = in->num_short_term_ref_pic_sets;
+    out->num_long_term_ref_pics_sps = in->num_long_term_ref_pics_sps;
+    out->pcm_sample_bit_depth_luma_minus1 = in->pcm_sample_bit_depth_luma_minus1;
+    out->pcm_sample_bit_depth_chroma_minus1 = in->pcm_sample_bit_depth_chroma_minus1;
+    out->log2_min_pcm_luma_coding_block_size_minus3 = in->log2_min_pcm_luma_coding_block_size_minus3;
+    out->log2_diff_max_min_pcm_luma_coding_block_size = in->log2_diff_max_min_pcm_luma_coding_block_size;
+    out->reserved1 = in->reserved1;
+    out->reserved2 = in->reserved2;
+    out->palette_max_size = in->palette_max_size;
+    out->delta_palette_max_predictor_size = in->delta_palette_max_predictor_size;
+    out->motion_vector_resolution_control_idc = in->motion_vector_resolution_control_idc;
+    out->sps_num_palette_predictor_initializers_minus1 = in->sps_num_palette_predictor_initializers_minus1;
+    out->conf_win_left_offset = in->conf_win_left_offset;
+    out->conf_win_right_offset = in->conf_win_right_offset;
+    out->conf_win_top_offset = in->conf_win_top_offset;
+    out->conf_win_bottom_offset = in->conf_win_bottom_offset;
+    out->pProfileTierLevel = UlongToPtr(in->pProfileTierLevel);
+    out->pDecPicBufMgr = UlongToPtr(in->pDecPicBufMgr);
+    out->pScalingLists = UlongToPtr(in->pScalingLists);
+    out->pShortTermRefPicSet = UlongToPtr(in->pShortTermRefPicSet);
+    out->pLongTermRefPicsSps = UlongToPtr(in->pLongTermRefPicsSps);
+    out->pSequenceParameterSetVui = convert_StdVideoH265SequenceParameterSetVui_array_win32_to_host(ctx, (const StdVideoH265SequenceParameterSetVui32 *)UlongToPtr(in->pSequenceParameterSetVui), 1);
+    out->pPredictorPaletteEntries = UlongToPtr(in->pPredictorPaletteEntries);
+}
+
+static const StdVideoH265SequenceParameterSet *convert_StdVideoH265SequenceParameterSet_array_win32_to_host(struct conversion_context *ctx, const StdVideoH265SequenceParameterSet32 *in, uint32_t count)
+{
+    StdVideoH265SequenceParameterSet *out;
+    unsigned int i;
+
+    if (!in || !count) return NULL;
+
+    out = conversion_context_alloc(ctx, count * sizeof(*out));
+    for (i = 0; i < count; i++)
+    {
+        convert_StdVideoH265SequenceParameterSet_win32_to_host(ctx, &in[i], &out[i]);
+    }
+
+    return out;
+}
+
+static void convert_StdVideoH265PictureParameterSet_win32_to_host(const StdVideoH265PictureParameterSet32 *in, StdVideoH265PictureParameterSet *out)
+{
+    if (!in) return;
+
+    out->flags = in->flags;
+    out->pps_pic_parameter_set_id = in->pps_pic_parameter_set_id;
+    out->pps_seq_parameter_set_id = in->pps_seq_parameter_set_id;
+    out->sps_video_parameter_set_id = in->sps_video_parameter_set_id;
+    out->num_extra_slice_header_bits = in->num_extra_slice_header_bits;
+    out->num_ref_idx_l0_default_active_minus1 = in->num_ref_idx_l0_default_active_minus1;
+    out->num_ref_idx_l1_default_active_minus1 = in->num_ref_idx_l1_default_active_minus1;
+    out->init_qp_minus26 = in->init_qp_minus26;
+    out->diff_cu_qp_delta_depth = in->diff_cu_qp_delta_depth;
+    out->pps_cb_qp_offset = in->pps_cb_qp_offset;
+    out->pps_cr_qp_offset = in->pps_cr_qp_offset;
+    out->pps_beta_offset_div2 = in->pps_beta_offset_div2;
+    out->pps_tc_offset_div2 = in->pps_tc_offset_div2;
+    out->log2_parallel_merge_level_minus2 = in->log2_parallel_merge_level_minus2;
+    out->log2_max_transform_skip_block_size_minus2 = in->log2_max_transform_skip_block_size_minus2;
+    out->diff_cu_chroma_qp_offset_depth = in->diff_cu_chroma_qp_offset_depth;
+    out->chroma_qp_offset_list_len_minus1 = in->chroma_qp_offset_list_len_minus1;
+    memcpy(out->cb_qp_offset_list, in->cb_qp_offset_list, STD_VIDEO_H265_CHROMA_QP_OFFSET_LIST_SIZE * sizeof(int8_t));
+    memcpy(out->cr_qp_offset_list, in->cr_qp_offset_list, STD_VIDEO_H265_CHROMA_QP_OFFSET_LIST_SIZE * sizeof(int8_t));
+    out->log2_sao_offset_scale_luma = in->log2_sao_offset_scale_luma;
+    out->log2_sao_offset_scale_chroma = in->log2_sao_offset_scale_chroma;
+    out->pps_act_y_qp_offset_plus5 = in->pps_act_y_qp_offset_plus5;
+    out->pps_act_cb_qp_offset_plus5 = in->pps_act_cb_qp_offset_plus5;
+    out->pps_act_cr_qp_offset_plus3 = in->pps_act_cr_qp_offset_plus3;
+    out->pps_num_palette_predictor_initializers = in->pps_num_palette_predictor_initializers;
+    out->luma_bit_depth_entry_minus8 = in->luma_bit_depth_entry_minus8;
+    out->chroma_bit_depth_entry_minus8 = in->chroma_bit_depth_entry_minus8;
+    out->num_tile_columns_minus1 = in->num_tile_columns_minus1;
+    out->num_tile_rows_minus1 = in->num_tile_rows_minus1;
+    out->reserved1 = in->reserved1;
+    out->reserved2 = in->reserved2;
+    memcpy(out->column_width_minus1, in->column_width_minus1, STD_VIDEO_H265_CHROMA_QP_OFFSET_TILE_COLS_LIST_SIZE * sizeof(uint16_t));
+    memcpy(out->row_height_minus1, in->row_height_minus1, STD_VIDEO_H265_CHROMA_QP_OFFSET_TILE_ROWS_LIST_SIZE * sizeof(uint16_t));
+    out->reserved3 = in->reserved3;
+    out->pScalingLists = UlongToPtr(in->pScalingLists);
+    out->pPredictorPaletteEntries = UlongToPtr(in->pPredictorPaletteEntries);
+}
+
+static const StdVideoH265PictureParameterSet *convert_StdVideoH265PictureParameterSet_array_win32_to_host(struct conversion_context *ctx, const StdVideoH265PictureParameterSet32 *in, uint32_t count)
+{
+    StdVideoH265PictureParameterSet *out;
+    unsigned int i;
+
+    if (!in || !count) return NULL;
+
+    out = conversion_context_alloc(ctx, count * sizeof(*out));
+    for (i = 0; i < count; i++)
+    {
+        convert_StdVideoH265PictureParameterSet_win32_to_host(&in[i], &out[i]);
+    }
+
+    return out;
+}
+
+static void convert_VkVideoDecodeH265SessionParametersAddInfoKHR_win32_to_host(struct conversion_context *ctx, const VkVideoDecodeH265SessionParametersAddInfoKHR32 *in, VkVideoDecodeH265SessionParametersAddInfoKHR *out)
+{
+    if (!in) return;
+
+    out->sType = in->sType;
+    out->pNext = NULL;
+    out->stdVPSCount = in->stdVPSCount;
+    out->pStdVPSs = convert_StdVideoH265VideoParameterSet_array_win32_to_host(ctx, (const StdVideoH265VideoParameterSet32 *)UlongToPtr(in->pStdVPSs), in->stdVPSCount);
+    out->stdSPSCount = in->stdSPSCount;
+    out->pStdSPSs = convert_StdVideoH265SequenceParameterSet_array_win32_to_host(ctx, (const StdVideoH265SequenceParameterSet32 *)UlongToPtr(in->pStdSPSs), in->stdSPSCount);
+    out->stdPPSCount = in->stdPPSCount;
+    out->pStdPPSs = convert_StdVideoH265PictureParameterSet_array_win32_to_host(ctx, (const StdVideoH265PictureParameterSet32 *)UlongToPtr(in->pStdPPSs), in->stdPPSCount);
+    if (in->pNext)
+        FIXME("Unexpected pNext\n");
+}
+
+static const VkVideoDecodeH265SessionParametersAddInfoKHR *convert_VkVideoDecodeH265SessionParametersAddInfoKHR_array_win32_to_host(struct conversion_context *ctx, const VkVideoDecodeH265SessionParametersAddInfoKHR32 *in, uint32_t count)
+{
+    VkVideoDecodeH265SessionParametersAddInfoKHR *out;
+    unsigned int i;
+
+    if (!in || !count) return NULL;
+
+    out = conversion_context_alloc(ctx, count * sizeof(*out));
+    for (i = 0; i < count; i++)
+    {
+        convert_VkVideoDecodeH265SessionParametersAddInfoKHR_win32_to_host(ctx, &in[i], &out[i]);
+    }
+
+    return out;
+}
+
 static void convert_VkVideoEncodeH264SessionParametersAddInfoKHR_win32_to_host(struct conversion_context *ctx, const VkVideoEncodeH264SessionParametersAddInfoKHR32 *in, VkVideoEncodeH264SessionParametersAddInfoKHR *out)
 {
     if (!in) return;
@@ -27685,6 +28201,20 @@ static void convert_VkVideoSessionParametersCreateInfoKHR_win32_to_host(struct c
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_SESSION_PARAMETERS_CREATE_INFO_KHR:
+        {
+            VkVideoDecodeH265SessionParametersCreateInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
+            const VkVideoDecodeH265SessionParametersCreateInfoKHR32 *in_ext = (const VkVideoDecodeH265SessionParametersCreateInfoKHR32 *)in_header;
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_SESSION_PARAMETERS_CREATE_INFO_KHR;
+            out_ext->pNext = NULL;
+            out_ext->maxStdVPSCount = in_ext->maxStdVPSCount;
+            out_ext->maxStdSPSCount = in_ext->maxStdSPSCount;
+            out_ext->maxStdPPSCount = in_ext->maxStdPPSCount;
+            out_ext->pParametersAddInfo = convert_VkVideoDecodeH265SessionParametersAddInfoKHR_array_win32_to_host(ctx, (const VkVideoDecodeH265SessionParametersAddInfoKHR32 *)UlongToPtr(in_ext->pParametersAddInfo), 1);
+            out_header->pNext = (void *)out_ext;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_ENCODE_AV1_SESSION_PARAMETERS_CREATE_INFO_KHR:
         {
             VkVideoEncodeAV1SessionParametersCreateInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
@@ -39260,6 +39790,15 @@ static void convert_VkVideoCapabilitiesKHR_win32_to_host(struct conversion_conte
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_CAPABILITIES_KHR:
+        {
+            VkVideoDecodeH265CapabilitiesKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_CAPABILITIES_KHR;
+            out_ext->pNext = NULL;
+            out_header->pNext = (void *)out_ext;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_DECODE_VP9_CAPABILITIES_KHR:
         {
             VkVideoDecodeVP9CapabilitiesKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
@@ -39406,6 +39945,15 @@ static void convert_VkVideoCapabilitiesKHR_host_to_win32(const VkVideoCapabiliti
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_CAPABILITIES_KHR:
+        {
+            VkVideoDecodeH265CapabilitiesKHR32 *out_ext = find_next_struct32(out_header, VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_CAPABILITIES_KHR);
+            const VkVideoDecodeH265CapabilitiesKHR *in_ext = (const VkVideoDecodeH265CapabilitiesKHR *)in_header;
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_CAPABILITIES_KHR;
+            out_ext->maxLevelIdc = in_ext->maxLevelIdc;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_DECODE_VP9_CAPABILITIES_KHR:
         {
             VkVideoDecodeVP9CapabilitiesKHR32 *out_ext = find_next_struct32(out_header, VK_STRUCTURE_TYPE_VIDEO_DECODE_VP9_CAPABILITIES_KHR);
@@ -42628,6 +43176,22 @@ static void convert_VkVideoSessionParametersUpdateInfoKHR_win32_to_host(struct c
             out_header = (void *)out_ext;
             break;
         }
+        case VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_SESSION_PARAMETERS_ADD_INFO_KHR:
+        {
+            VkVideoDecodeH265SessionParametersAddInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
+            const VkVideoDecodeH265SessionParametersAddInfoKHR32 *in_ext = (const VkVideoDecodeH265SessionParametersAddInfoKHR32 *)in_header;
+            out_ext->sType = VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_SESSION_PARAMETERS_ADD_INFO_KHR;
+            out_ext->pNext = NULL;
+            out_ext->stdVPSCount = in_ext->stdVPSCount;
+            out_ext->pStdVPSs = convert_StdVideoH265VideoParameterSet_array_win32_to_host(ctx, (const StdVideoH265VideoParameterSet32 *)UlongToPtr(in_ext->pStdVPSs), in_ext->stdVPSCount);
+            out_ext->stdSPSCount = in_ext->stdSPSCount;
+            out_ext->pStdSPSs = convert_StdVideoH265SequenceParameterSet_array_win32_to_host(ctx, (const StdVideoH265SequenceParameterSet32 *)UlongToPtr(in_ext->pStdSPSs), in_ext->stdSPSCount);
+            out_ext->stdPPSCount = in_ext->stdPPSCount;
+            out_ext->pStdPPSs = convert_StdVideoH265PictureParameterSet_array_win32_to_host(ctx, (const StdVideoH265PictureParameterSet32 *)UlongToPtr(in_ext->pStdPPSs), in_ext->stdPPSCount);
+            out_header->pNext = (void *)out_ext;
+            out_header = (void *)out_ext;
+            break;
+        }
         case VK_STRUCTURE_TYPE_VIDEO_ENCODE_H264_SESSION_PARAMETERS_ADD_INFO_KHR:
         {
             VkVideoEncodeH264SessionParametersAddInfoKHR *out_ext = conversion_context_alloc(ctx, sizeof(*out_ext));
diff --git a/include/wine/vulkan.h b/include/wine/vulkan.h
index 7897ffd5..cca61ba5 100644
--- a/include/wine/vulkan.h
+++ b/include/wine/vulkan.h
@@ -344,6 +344,8 @@ typedef struct _XDisplay Display;
 #define VK_EXT_CALIBRATED_TIMESTAMPS_EXTENSION_NAME "VK_EXT_calibrated_timestamps"
 #define VK_AMD_SHADER_CORE_PROPERTIES_SPEC_VERSION 2
 #define VK_AMD_SHADER_CORE_PROPERTIES_EXTENSION_NAME "VK_AMD_shader_core_properties"
+#define VK_KHR_VIDEO_DECODE_H265_SPEC_VERSION 8
+#define VK_KHR_VIDEO_DECODE_H265_EXTENSION_NAME "VK_KHR_video_decode_h265"
 #define VK_MAX_GLOBAL_PRIORITY_SIZE_KHR VK_MAX_GLOBAL_PRIORITY_SIZE
 #define VK_KHR_GLOBAL_PRIORITY_SPEC_VERSION 1
 #define VK_KHR_GLOBAL_PRIORITY_EXTENSION_NAME "VK_KHR_global_priority"
@@ -5721,6 +5723,12 @@ typedef enum VkStructureType
     VK_STRUCTURE_TYPE_PIPELINE_COMPILER_CONTROL_CREATE_INFO_AMD = 1000183000,
     VK_STRUCTURE_TYPE_CALIBRATED_TIMESTAMP_INFO_KHR = 1000184000,
     VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_CORE_PROPERTIES_AMD = 1000185000,
+    VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_CAPABILITIES_KHR = 1000187000,
+    VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_SESSION_PARAMETERS_CREATE_INFO_KHR = 1000187001,
+    VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_SESSION_PARAMETERS_ADD_INFO_KHR = 1000187002,
+    VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PROFILE_INFO_KHR = 1000187003,
+    VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_PICTURE_INFO_KHR = 1000187004,
+    VK_STRUCTURE_TYPE_VIDEO_DECODE_H265_DPB_SLOT_INFO_KHR = 1000187005,
     VK_STRUCTURE_TYPE_DEVICE_MEMORY_OVERALLOCATION_CREATE_INFO_AMD = 1000189000,
     VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VERTEX_ATTRIBUTE_DIVISOR_PROPERTIES_EXT = 1000190000,
     VK_STRUCTURE_TYPE_PIPELINE_VERTEX_INPUT_DIVISOR_STATE_CREATE_INFO = 1000190001,
@@ -6861,6 +6869,7 @@ typedef enum VkVideoCodecOperationFlagBitsKHR
 {
     VK_VIDEO_CODEC_OPERATION_NONE_KHR = 0,
     VK_VIDEO_CODEC_OPERATION_DECODE_H264_BIT_KHR = 0x00000001,
+    VK_VIDEO_CODEC_OPERATION_DECODE_H265_BIT_KHR = 0x00000002,
     VK_VIDEO_CODEC_OPERATION_DECODE_AV1_BIT_KHR = 0x00000004,
     VK_VIDEO_CODEC_OPERATION_DECODE_VP9_BIT_KHR = 0x00000008,
     VK_VIDEO_CODEC_OPERATION_ENCODE_H264_BIT_KHR = 0x00010000,
@@ -10267,6 +10276,18 @@ typedef struct VkVideoDecodeH264SessionParametersAddInfoKHR
     const StdVideoH264PictureParameterSet *pStdPPSs;
 } VkVideoDecodeH264SessionParametersAddInfoKHR;
 
+typedef struct VkVideoDecodeH265SessionParametersAddInfoKHR
+{
+    VkStructureType sType;
+    const void *pNext;
+    uint32_t stdVPSCount;
+    const StdVideoH265VideoParameterSet *pStdVPSs;
+    uint32_t stdSPSCount;
+    const StdVideoH265SequenceParameterSet *pStdSPSs;
+    uint32_t stdPPSCount;
+    const StdVideoH265PictureParameterSet *pStdPPSs;
+} VkVideoDecodeH265SessionParametersAddInfoKHR;
+
 typedef struct VkVideoEncodeAV1FrameSizeKHR
 {
     uint32_t intraFrameSize;
@@ -18539,6 +18560,46 @@ typedef struct VkVideoDecodeH264SessionParametersCreateInfoKHR
     const VkVideoDecodeH264SessionParametersAddInfoKHR *pParametersAddInfo;
 } VkVideoDecodeH264SessionParametersCreateInfoKHR;
 
+typedef struct VkVideoDecodeH265CapabilitiesKHR
+{
+    VkStructureType sType;
+    void *pNext;
+    StdVideoH265LevelIdc maxLevelIdc;
+} VkVideoDecodeH265CapabilitiesKHR;
+
+typedef struct VkVideoDecodeH265DpbSlotInfoKHR
+{
+    VkStructureType sType;
+    const void *pNext;
+    const StdVideoDecodeH265ReferenceInfo *pStdReferenceInfo;
+} VkVideoDecodeH265DpbSlotInfoKHR;
+
+typedef struct VkVideoDecodeH265PictureInfoKHR
+{
+    VkStructureType sType;
+    const void *pNext;
+    const StdVideoDecodeH265PictureInfo *pStdPictureInfo;
+    uint32_t sliceSegmentCount;
+    const uint32_t *pSliceSegmentOffsets;
+} VkVideoDecodeH265PictureInfoKHR;
+
+typedef struct VkVideoDecodeH265ProfileInfoKHR
+{
+    VkStructureType sType;
+    const void *pNext;
+    StdVideoH265ProfileIdc stdProfileIdc;
+} VkVideoDecodeH265ProfileInfoKHR;
+
+typedef struct VkVideoDecodeH265SessionParametersCreateInfoKHR
+{
+    VkStructureType sType;
+    const void *pNext;
+    uint32_t maxStdVPSCount;
+    uint32_t maxStdSPSCount;
+    uint32_t maxStdPPSCount;
+    const VkVideoDecodeH265SessionParametersAddInfoKHR *pParametersAddInfo;
+} VkVideoDecodeH265SessionParametersCreateInfoKHR;
+
 typedef struct VkVideoDecodeInfoKHR
 {
     VkStructureType sType;
@@ -21472,6 +21533,7 @@ VkResult VKAPI_CALL vkWriteMicromapsPropertiesEXT(VkDevice device, uint32_t micr
     USE_VK_EXT(VK_KHR_vertex_attribute_divisor) \
     USE_VK_EXT(VK_KHR_video_decode_av1) \
     USE_VK_EXT(VK_KHR_video_decode_h264) \
+    USE_VK_EXT(VK_KHR_video_decode_h265) \
     USE_VK_EXT(VK_KHR_video_decode_queue) \
     USE_VK_EXT(VK_KHR_video_decode_vp9) \
     USE_VK_EXT(VK_KHR_video_encode_av1) \
-- 
2.55.0

